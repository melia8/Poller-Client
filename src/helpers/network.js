export async function fetchData(url) {
  try {
    const response = await fetch(url);
    const jsonData = await response.json();
    return jsonData;
  } catch (error) {
    console.error("Error fetching data:", error);
  }
}

export async function sendData(url, body) {
  try {
    const response = await fetch(url, {
      method: "POST",
      body: JSON.stringify(body),
      headers: {
        "Content-Type": "application/json",
      },
    });
    const jsonData = await response.json();
    return jsonData;
  } catch (error) {
    console.error("Error fetching data:", error);
  }
}
