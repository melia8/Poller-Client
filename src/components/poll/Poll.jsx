import PollBar from "./PollBar.jsx";
import classes from "./Poll.module.css";
import { useEffect, useState } from "react";
import { fetchData } from "../../helpers/network.js";
import Header from "../header/Header.jsx";

const livePollUrl = "http://localhost:8080/poll";

function Poll() {
  const [pollData, setPollData] = useState({});

  useEffect(() => {
    fetchData(livePollUrl).then((data) => {
      setPollData(data);
    });
  }, []);

  return (
    <div className={classes.layout}>
      <Header title={pollData.title} />
      {pollData.options &&
        pollData.options.map((data) => (
          <PollBar
            key={data.id}
            value={data.total}
            total={pollData.total}
            text={data.name}
          />
        ))}
    </div>
  );
}

export default Poll;
