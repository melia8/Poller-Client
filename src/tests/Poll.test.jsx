import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import Poll from "../components/poll/Poll";
import { act } from "react";
import getTestData from "./testData";
import { fetchData } from "../helpers/network";

vi.mock("../helpers/network.js", () => ({
  fetchData: vi.fn(),
}));

fetchData.mockImplementation(async () => {
  return getTestData();
});

describe("Poll", async () => {
  it("The component UI should not change", async () => {
    const component = await act(async () => render(<Poll />));
    expect(component).toMatchSnapshot();
  });

  it("renders correct percentage", async () => {
    await act(async () => render(<Poll />));

    const getPercent = (team) =>
      screen.getByText(team).nextSibling.textContent.trim();

    expect(getPercent("Liverpool")).toBe("50.0%");
    expect(getPercent("Manchester City")).toBe("30.0%");
    expect(getPercent("Arsenal")).toBe("20.0%");
    expect(getPercent("Manchester United")).toBe("6.00%");
    expect(getPercent("West Ham United")).toBe("4.00%");
  });
});
