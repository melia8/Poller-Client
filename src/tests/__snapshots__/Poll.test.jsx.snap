// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Poll > The component UI should not change 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_layout_49ffbd"
      >
        <div
          class="_logo_69e643"
        >
          <a
            class="_logoLink_69e643"
            href="https://dizplai.com/home/"
            rel="nofollow"
          >
            <img
              alt="Logo"
              height="60"
              src="https://dizplai.com/wp-content/themes/neverno/library/images/logo.svg"
              width="124"
            />
          </a>
        </div>
        <div
          class="_title_69e643"
        >
           
          Who will win Premier League
        </div>
        <div
          class="_progressBarBackground_a525df"
        >
          <div
            class="_title_a525df"
          >
             
            Liverpool
             
          </div>
          <div
            class="_percent_a525df"
          >
             
            50.0
            % 
          </div>
          <div
            class="_progressBar_a525df"
            style="width: 50.0%;"
          />
        </div>
        <div
          class="_progressBarBackground_a525df"
        >
          <div
            class="_title_a525df"
          >
             
            Manchester City
             
          </div>
          <div
            class="_percent_a525df"
          >
             
            30.0
            % 
          </div>
          <div
            class="_progressBar_a525df"
            style="width: 30.0%;"
          />
        </div>
        <div
          class="_progressBarBackground_a525df"
        >
          <div
            class="_title_a525df"
          >
             
            Arsenal
             
          </div>
          <div
            class="_percent_a525df"
          >
             
            20.0
            % 
          </div>
          <div
            class="_progressBar_a525df"
            style="width: 20.0%;"
          />
        </div>
        <div
          class="_progressBarBackground_a525df"
        >
          <div
            class="_title_a525df"
          >
             
            Manchester United
             
          </div>
          <div
            class="_percent_a525df"
          >
             
            6.00
            % 
          </div>
          <div
            class="_progressBar_a525df"
            style="width: 6.00%;"
          />
        </div>
        <div
          class="_progressBarBackground_a525df"
        >
          <div
            class="_title_a525df"
          >
             
            West Ham United
             
          </div>
          <div
            class="_percent_a525df"
          >
             
            4.00
            % 
          </div>
          <div
            class="_progressBar_a525df"
            style="width: 4.00%;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_layout_49ffbd"
    >
      <div
        class="_logo_69e643"
      >
        <a
          class="_logoLink_69e643"
          href="https://dizplai.com/home/"
          rel="nofollow"
        >
          <img
            alt="Logo"
            height="60"
            src="https://dizplai.com/wp-content/themes/neverno/library/images/logo.svg"
            width="124"
          />
        </a>
      </div>
      <div
        class="_title_69e643"
      >
         
        Who will win Premier League
      </div>
      <div
        class="_progressBarBackground_a525df"
      >
        <div
          class="_title_a525df"
        >
           
          Liverpool
           
        </div>
        <div
          class="_percent_a525df"
        >
           
          50.0
          % 
        </div>
        <div
          class="_progressBar_a525df"
          style="width: 50.0%;"
        />
      </div>
      <div
        class="_progressBarBackground_a525df"
      >
        <div
          class="_title_a525df"
        >
           
          Manchester City
           
        </div>
        <div
          class="_percent_a525df"
        >
           
          30.0
          % 
        </div>
        <div
          class="_progressBar_a525df"
          style="width: 30.0%;"
        />
      </div>
      <div
        class="_progressBarBackground_a525df"
      >
        <div
          class="_title_a525df"
        >
           
          Arsenal
           
        </div>
        <div
          class="_percent_a525df"
        >
           
          20.0
          % 
        </div>
        <div
          class="_progressBar_a525df"
          style="width: 20.0%;"
        />
      </div>
      <div
        class="_progressBarBackground_a525df"
      >
        <div
          class="_title_a525df"
        >
           
          Manchester United
           
        </div>
        <div
          class="_percent_a525df"
        >
           
          6.00
          % 
        </div>
        <div
          class="_progressBar_a525df"
          style="width: 6.00%;"
        />
      </div>
      <div
        class="_progressBarBackground_a525df"
      >
        <div
          class="_title_a525df"
        >
           
          West Ham United
           
        </div>
        <div
          class="_percent_a525df"
        >
           
          4.00
          % 
        </div>
        <div
          class="_progressBar_a525df"
          style="width: 4.00%;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
